src/DotNetWorker.Core/Context/DefaultFunctionContext.cs

```diff
@@ -58,11 +58,11 @@ public override IServiceProvider InstanceServices
 
         public override RetryContext RetryContext => Features.GetRequired<IExecutionRetryFeature>().Context;
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            if (_instanceServicesScope != null)
+            if (_instanceServicesScope is IAsyncDisposable asyncServiceScope)
```

Comment:
Similar concern here, as we don't control the scope implementation coming back and that might only implement `IDisposable` which would cause scoped services to no longer be disposed of with the changes.

Reply:
Done.